src/platform_impl/windows/dark_mode.rs

```diff
@@ -132,11 +124,12 @@ fn set_dark_mode_for_window(hwnd: HWND, is_dark_mode: bool) {
                 cbData: std::mem::size_of_val(&is_dark_mode_bigbool) as _,
             };
 
-            assert_eq!(
-                1,
-                set_window_composition_attribute(hwnd, &mut data as *mut _)
-            );
+            let status = set_window_composition_attribute(hwnd, &mut data as *mut _);
+
+            status == 0
```

Comment:
```suggestion
            status == 1
```
Actually it would also be nice to find out what is making this crash:
```suggestion
            match status {
                1 => true,
                error => {
                    log::error!("{}: {}", error, std::io::Error::last_os_error());
                    false
                }
            }
```

Reply:
Just a wild guess, but I'd assume the `WCA_USEDARKMODECOLORS` window compositing attribute is unknown. That's really the only thing that could fail here.

Reply:
Yeah I assume so too, would be nice to know for sure though.
In any case, `0` is wrong, it should be `1`, accord to [this](https://www.undoc.airesoft.co.uk/user32.dll/SetWindowCompositionAttribute.php):
> Nonzero on success, zero otherwise. Call GetLastError on failure for more information.

Reply:
Oh yeah, my bad. I previously have returned a bool on `status == 0` here and have just changed the statement without inverting the assertion. Great catch.

Reply:
It should be `status != 0` though, not `status == 1`.

Reply:
Yeah the documentation is actually not accurate, the return type is [`BOOL`](https://docs.rs/winapi/0.3.8/winapi/shared/minwindef/type.BOOL.html), which can only be [`TRUE`](https://docs.rs/winapi/0.3.8/winapi/shared/minwindef/constant.TRUE.html)(1) or [`FALSE`](https://docs.rs/winapi/0.3.8/winapi/shared/minwindef/constant.FALSE.html)(0), we should actually check for that.

```suggestion
            status == TRUE
```
And also add `use winapi::shared::minwindef::{TRUE, FALSE};`

Reply:
I mean that's basically how booleans work, usually it's 1 for true and 0 for false, but all other values should also be treated as true. That's probably why the documentation states it that way, so checking for `!= 0` means you're on the safe side.